example/router/router.hpp

```diff
@@ -0,0 +1,260 @@
+    router_base();
+
+    BOOST_URL_DECL
+    void
+    route_impl(
+        string_view s,
+        std::size_t idx,
+        std::size_t maxn);
+
+    BOOST_URL_DECL
+    node const*
+    match_impl(
+        string_view s,
+        string_view*& matches,
+        string_view*& names) const noexcept;
+};
+
+
+/** A URL router.
+
+    This container matches static and dynamic
+    URL requests to an object which represents
+    how the it should be handled. These
+    values are usually callback functions.
+
+    @tparam T type of resource associated with
+    each path template
+
+    @tparam N maximum number of replacement fields
+    in a path template
+
+    @par Exception Safety
+
+    @li Functions marked `noexcept` provide the
+    no-throw guarantee, otherwise:
+
+    @li Functions which throw offer the strong
+    exception safety guarantee.
+
+    @see
+        @ref parse_absolute_uri,
+        @ref parse_relative_ref,
+        @ref parse_uri,
+        @ref parse_uri_reference,
+        @ref resolve.
+*/
+template <class T, std::size_t N = 20>
+class router : public router_base
+{
+public:
+    /// A range type with the match results
+    class match_results
+        : public match_results_base
+    {
+        string_view matches_storage_[N];
+        string_view ids_storage_[N];
+        T const* data_{nullptr};
+
+        friend router;
+
+        match_results() = default;
+
+        match_results(
+            T const* data,
+            node const* leaf,
+            string_view matches[N],
+            string_view ids[N],
+            std::size_t n)
+            : match_results_base(
+                leaf, matches_storage_, ids_storage_, n)
+            , data_(data)
+        {
+            for (std::size_t i = 0; i < n; ++i)
+            {
+                matches_storage_[i] = matches[i];
+                ids_storage_[i] = ids[i];
+            }
+        }
+
+    public:
+        T const&
+        operator*() const;
+
+        T const&
+        value() const;
+    };
+
+    /// Constructor
+    router() = default;
+
+    /** Route the specified URL path to a resource
+
+        @param path A url path with dynamic segments
+        @param resource A resource the path corresponds to
+
+        @see
+            https://fmt.dev/latest/syntax.html
+     */
+    template <class U>
+    void
+    route(string_view path, U&& resource);
+
+    /** Match URL path to corresponding resource
+
+        @param request Request path
+        @return The match results
+     */
+    match_results
+    match(string_view request) const noexcept;
+
+    /** Match URL path and stores results into args
+
+        @param request Request path
+        @param args Storage for the matches
+
+        @return True if we found a match
+     */
+    template <class ...Args>
+    result<T>
+    match_to(string_view request, Args&... args) const;
+
+private:
+    std::vector<T> data_;
+};
+
+} // urls
+} // boost
+
+#include <impl/router.hpp>
```

Comment:
Should not be in angle brackets since it is a relative path

Reply:
I put `.` in include paths. I got that from another example but I don't remember which.

Reply:
I mean just put it in double quotes